aws/data_source_aws_connect_instance.go

```diff
@@ -0,0 +1,142 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/connect"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var dataResourceConnectInstanceAttributesMapping = map[string]string{
+	connect.InstanceAttributeTypeInboundCalls:          "inbound_calls_enabled",
+	connect.InstanceAttributeTypeOutboundCalls:         "outbound_calls_enabled",
+	connect.InstanceAttributeTypeContactflowLogs:       "contact_flow_logs_enabled",
+	connect.InstanceAttributeTypeContactLens:           "contact_lens_enabled",
+	connect.InstanceAttributeTypeAutoResolveBestVoices: "auto_resolve_best_voices",
+	connect.InstanceAttributeTypeUseCustomTtsVoices:    "use_custom_tts_voices",
+	connect.InstanceAttributeTypeEarlyMedia:            "early_media_enabled",
+}
+
+func dataSourceAwsConnectInstance() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceAwsConnectInstanceRead,
+		Schema: map[string]*schema.Schema{
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"created_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"identity_management_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"instance_alias": {
```

Comment:
can we also have data source fetch by alias as well

Reply:
for some source checkout https://github.com/thornleyk/terraform-provider-aws/blob/f-aws_connect_instance/aws/data_source_aws_connect_instance.go

Reply:
That's a good idea as well! I can update this to have the data source work with both

Reply:
added the option to lookup by both id & alias. lmk wdyt?